components/brave_extension/extension/brave_extension/background/greaselion.ts

```diff
@@ -164,28 +217,47 @@ const handleRegisterOnSendHeadersWebRequest = (registrationKey: string, mediaTyp
   // Mark this tab as registered
   onSendHeadersWebRequestRegistrations.set(registrationKey, { registered: true })
 
-  chrome.webRequest.onSendHeaders.addListener(
-    // Listener
-    function (details) {
-      const port = portsByTabIdSenderId.get(registrationKey)
-      if (!port) {
-        return
-      }
-      port.postMessage({
-        type: 'OnSendHeadersWebRequest',
-        mediaType,
-        data: {
-          details
-        }
-      })
-    },
-    // Filters
-    {
-      urls: data.urlPatterns
-    },
-    // Extra
-    data.extra
-  )
+  // Create and install the listener
+  const connectionState = connectionsByTabIdSenderId.get(registrationKey)
+  if (!connectionState) {
+    return
+  }
+  connectionState.onSendHeadersWebRequestListener =
+    onSendHeadersWebRequest.bind(null, registrationKey, mediaType)
+  if (connectionState.onSendHeadersWebRequestListener) {
```

Comment:
I appreciate the defensive programming here, but is there a scenario where `Function.bind` would not return a newly bound function?

Reply:
Unfortunately, when I didn't have that there Typescript was complaining because `onSendHeadersWebRequestListener` is an optional property in the interface. This seemed like the best way around that, but open to other suggestions too.

Reply:
Maybe introduce a local variable for the listener? (e.g. `const listener = X.bind(...)`)

Reply:
That works, thanks!